Controller/PurchaseController.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+/**
+ * @author Rafał Muszyński <rafal.muszynski@sourcefabric.org>
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.gnu.org/licenses/gpl-3.0.txt
+ */
+namespace Newscoop\PaywallBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Newscoop\PaywallBundle\Entity\Payment;
+use Newscoop\PaywallBundle\Events\PaywallEvents;
+use Newscoop\PaywallBundle\Entity\OrderInterface;
+
+/**
+ * It handles actions after payment redirects.
+ */
+class PurchaseController extends BaseController
+{
+    /**
+     * @Route("/paywall/purchase/", name="paywall_plugin_purchase_purchase", options={"expose"=true})
+     *
+     * @Method("POST")
+     */
+    public function purchaseAction(Request $request)
+    {
+        $currencyProvider = $this->get('newscoop_paywall.currency_provider');
+        $currencyContext = $this->get('newscoop_paywall.currency_context');
+        $currencyContext->setCurrency($currencyProvider->getDefaultCurrency()->getCode());
+        $items = $request->request->get('batchorder', array());
+
+        if (empty($items)) {
+            return new Response($templatesService->fetchTemplate(
+                '_paywall/error.tpl',
+                array('msg' => 'no items')
+            ), 200, array('Content-Type' => 'text/html'));
```

Comment:
yes, as we want to render template and return some message there, so in this case 200 is appropriate code imo